backend/layers/business/business.py

```diff
@@ -403,6 +403,12 @@ def get_all_published_datasets(self) -> Iterable[DatasetVersion]:
         """
         return self.database_provider.get_all_datasets()
 
+    def get_all_published_collections_with_datasets(self) -> Iterable[CollectionVersionWithDatasets]:
+        """
+        Retrieves all the datasets from the database that belong to a published collection
+        """
+        return self.database_provider.get_all_published_collections_with_datasets()
```

Comment:
The persistence method should be called `get_all_mapped_collection_versions_with_datasets`, to be consistent with the distinction we've made in other methods between published collections (business concept) and mapped collection versions (schema concept). Mirrors 'get_all_mapped_collection_versions', and more clearly defines difference between that method and this one (and thus, the need for a different persistence method). 


note: while fields like 'is_published' exist in the db, those stay set to True once published even if the collection version is no longer the 'active' one. What we refer to in the business sense as 'published' usually means 'actively, currently published' which in the DB refers to whatever version is mapped to the canonical